### PR TITLE
Leapp upgrading and improvements

### PR DESCRIPTION
**Pull request by Lennonka**
I focused on improvements in the TOC, overall structure and content of upgrade paths.
We intend proper modularization later on.

* [x] I am familiar with the [contributing](CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3

### CODE REVIEW

File: guides/doc-Upgrading_and_Updating/master.adoc
Comment (Lennonka) [inline guides/doc-Upgrading_and_Updating/master.adoc:93]: What about foreman-deb and orcharhino?
cc @maximiliankolb

File: guides/doc-Upgrading_and_Updating/master.adoc
Comment (Lennonka) [inline guides/doc-Upgrading_and_Updating/master.adoc:97]: Again, what about foreman-deb and orcharhino?
cc @maximiliankolb

File: guides/doc-Upgrading_and_Updating/master.adoc
Comment (evgeni) [reply guides/doc-Upgrading_and_Updating/master.adoc:93]: foreman-deb can't be upgraded via Leapp

File: guides/doc-Upgrading_and_Updating/master.adoc
Comment (evgeni) [reply guides/doc-Upgrading_and_Updating/master.adoc:97]: That technically would work for foreman-deb, but I don't think the current wording works for non-EnterpriseLinux systems

Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/master.adoc:97]: So an upgrade to EL8 is not required for foreman-deb? Or we need to figure something else out for the intro.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (Lennonka) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:61]: @evgeni What should we mention in this place for non-EL systems?

File: guides/doc-Upgrading_and_Updating/master.adoc
Comment (evgeni) [reply guides/doc-Upgrading_and_Updating/master.adoc:97]: Correct. foreman-deb runs on Debian, not on Enterprise Linux.

**Comment by maximiliankolb** [comment]
For now, you don't have to include orcharhino. We cannot use it anyway if it's not modular. I'd appreciate it if this was prioritized on your end. Else, I will surely do it some day :)

**Comment by evgeni** [comment]
FWIW, the following patch makes it build for me:

```patch
diff --git guides/doc-Upgrading_and_Updating/master.adoc guides/doc-Upgrading_and_Updating/master.adoc
index 503bf235..935acd50 100644
--- guides/doc-Upgrading_and_Updating/master.adoc
+++ guides/doc-Upgrading_and_Updating/master.adoc
@@ -12,7 +12,9 @@ The process of advancing your {ProjectServer} and {SmartProxyServer} installatio
 For more information, see xref:upgrading_process_overview[].
 Updating::
 The process of advancing your {ProjectServer} and {SmartProxyServer} installations from a z-stream release to the next, for example {Project} {ProjectVersion}.0 to {Project} {ProjectVersion}.1.
+ifdef::foreman-el,katello,satellite[]
 For more information, see xref:updating_project[].
+endif::[]
 Migrating::
 The process of moving an existing {Project} installation to another hardware.
 ifdef::foreman-el,katello,satellite[]
diff --git guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
index d173e51e..ffbd2e96 100644
--- guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
+++ guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
@@ -11,7 +11,9 @@ For more information, see https://access.redhat.com/labs/satelliteupgradehelper/
 endif::[]
 
 Note that you can upgrade {SmartProxies} separately from {Project}.
+ifdef::foreman-el,katello,satellite[]
 For more information, see xref:Upgrading_Proxies_Separately_from_Server_{context}[].
+endif::[]
 
 [[upgrading_prerequisites]]
 == Prerequisites
@@ -57,8 +59,10 @@ endif::[]
 
 There are two ways of upgrading your OS:
 
+ifdef::foreman-el,katello,satellite[]
 * xref:upgrading-project-in-place-using-leapp_{context}[]
 * xref:migrating-project-to-el8_{context}[]
+endif:[]
 
 === Upgrading {ProjectName}
 
```

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:3]:
```suggestion
= Upgrading {Project} to Enterprise Linux 8 in-place using Leapp
```

feels weird to capitalize every word, or is it just me? (same below, obv)

File: guides/doc-Upgrading_and_Updating/topics/upgrading_content_hosts.adoc
Comment (evgeni) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_content_hosts.adoc:1]: renaming this topic and all the surrounding changes seem rather out of scope in this PR and make reading the diff harder :(

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (evgeni) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:90]: having this picture in the middle of the "high level steps" seems weird to me. it's a detail that doesn't *really* influence the high level steps (those are, in both cases, "upgrade satellite first", "then upgrade capsules")

File: guides/common/modules/con_migrating-project-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/con_migrating-project-to-el8.adoc:6]: Should this read "Migrating Project to a new Enterprise Linux system", to emphasize that you're moving to a different machine?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (evgeni) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:56]: I'm also unsure I like how you split up the "Upgrade Paths" section now, but don't have a good alternative suggestion right now. Let me ramble a bit, and maybe you can distill the right things from my ramblings? :sweat_smile: 

* As noted in the section below, the user needs to be on `{ProjectVersionPrevious}` to be able to upgrade to `{ProjectVersion}` -- we don't support skipping releases.
* Once they are on `{ProjectVersionPrevious}`, they can do the "normal" upgrade as described in `== Upgrading {ProjectName}` chapter (ch2 upstream, ch3 downstream)
* At this point, they are running the latest available version and are "done" with the version upgrade
* They *can* now continue with the OS upgrade, but don't have to -- this can happen at any later point (as long as it happens before they decide to upgrade to a major version that does not support OS upgrades anymore, i.e. Foreman 3.4)
* They *also can* perform minor updates (so Foreman 3.3.1, 3.3.2 etc, as in the very last chapter of the update guide), while still on EL7.

I am sure there will be users who decide to do both jumps "at once", and also those who will just update to the latest version and schedule the OS upgrade at a later point.

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc
Comment (Lennonka) [reply guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:3]: It's you, you don't know enough about capitalization in English. Leave this to writers experts, please :D

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:90]: So where would you like it better?

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc
Comment (evgeni) [reply guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:3]: k

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (evgeni) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:90]: If I would be writing this doc fresh (I am not), I wouldn't have included this picture in this paragraph at all.

But it was included and I don't have enough background knowledge to know why exactly, so I would probably just not move it (so leave it above the "The high level steps in upgrading" line and the "no upgrades from beta" warning)

Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:90]: I'll comment it out and we can get back to it later.

File: guides/common/modules/ref_post-upgrade-tasks.adoc
Comment (maximiliankolb) [inline guides/common/modules/ref_post-upgrade-tasks.adoc:2]: title does not match anchor/file name.

File: guides/doc-Upgrading_and_Updating/master.adoc
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/master.adoc:17]: Maybe "host" instead of hardware?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:63]: Missing anchor.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:68]: Can this link to `{SomethingDocURL}`?

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc
Comment (lpramuk) [reply guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:3]: In headings only prepositions and connectives are lower case.  I know it from song names :-)

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (lpramuk) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:56]: The sentence will be valid for 6.12 

For 6.11 it is vice versa like @evgeni said

File: guides/doc-Upgrading_and_Updating/master.adoc
Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/master.adoc:17]: Not a good idea, because it's about migrating the whole project, not a host

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:56]: Do they have to be on the latest .z release to be able to upgrade?

File: guides/doc-Upgrading_and_Updating/master.adoc
Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/master.adoc:17]: Maybe "a new instance" instead of "another hardware"?

File: guides/common/modules/con_migrating-project-to-a-new-el-system.adoc
Comment (evgeni) [inline guides/common/modules/con_migrating-project-to-a-new-el-system.adoc:2]: for Satellite, this probably should read "… to a New {RHEL} System"?
(We're trying to avoid RH naming for upstream, because it also runs on non-RHEL, but for Satellite it should be RHEL only)

File: guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc
Comment (evgeni) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:80]: Can we word this in a way, that it's clear that the upgrade to EL8 is not mandatory at the same *time*?

Maybe "you are able to upgrade"?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc
Comment (evgeni) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:83]: Same-ish comment as in the Satellite case, but with the added complexity that upstream users might already be on EL8 (I had forgotten about this detail when talking to you earlier).

So, maybe like this?
After you upgrade your {Project} *and it's running on Enterprise Linux 7*, you *can* upgrade the operating system on your {ProjectServer}s and {SmartProxies} to Enterprise Linux 8.

**Review by evgeni — COMMENTED** [review]
couple of minor comments, but overall :+1:

File: guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc
Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:80]: ah, sure thing, I thought that it was mandatory

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:54]: We would have to define DocURLs for pervious versions and it's for a greater debate whether that much work is actually benefical. If you want to consider it, please file an issue.

**Review by evgeni — APPROVED** [review]
content is :+1:

**Comment by evgeni** [comment]
ACK.

There is a similar-ish section (but for other dependency issues) in leapp.adoc:

```
ifdef::foreman-el,katello[]
+
The `preupgrade` might fail with a dependency resolution error such as:
…
```

The error people will need to act on in our case is:
```
  - cannot install the best update candidate for package ansible-2.9.27-1.el7ae.noarch
  - cannot install the best update candidate for package ansible-test-2.9.27-1.el7ae.noarch
```

**Comment by Lennonka** [comment]
I've decided to remind the user to review Known Issues in the RN instead copying them to the procedure. I'm considering it done.

@maximiliankolb Can you please give it a final formal check?

**Comment by Lennonka** [comment]
We might do the same with upstream RN if someone wants to take initiative.

**Comment by evgeni** [comment]
Works for me!

**Comment by maximiliankolb** [comment]
Will have a look at this on Friday.

**Comment by Lennonka** [comment]
Rebased.

File: guides/common/modules/ref_post-upgrade.adoc
Comment (maximiliankolb) [inline guides/common/modules/ref_post-upgrade.adoc:2]: The title is IHMO a bit too generic. Maybe "Post Leapp Update [Steps|Information]"?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:100]: Is this a comment block? https://docs.asciidoctor.org/asciidoc/latest/comments/#comment-blocks

If so, maybe remove it before merging.

**Review by maximiliankolb — APPROVED** [review]
The structure is not ideal yet, but it's all contained within `topics/`, so I am fine with it for now.

**Comment by vferschm** [comment]
Hi, I can see that some links are broken. The RHEL team really changed them recently:
Original:  _rhel_ 
New format:  _rhel_8_
Example: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/performing_a_standard_rhel_8_installation/index

**Comment by evgeni** [comment]
@vferschm I had opened https://github.com/theforeman/foreman-documentation/pull/1390 for that already

File: guides/common/modules/ref_post-upgrade.adoc
Comment (evgeni) [reply guides/common/modules/ref_post-upgrade.adoc:2]: Those steps do apply for non-leapp upgrades too

**Comment by evgeni** [comment]
Lena is out, so I am going to merge this and @vferschm can coordinate on any post-merge cleanups ;)

**Comment by evgeni** [comment]
bbf96b7cd..298d00ffe  3.3 -> 3.3

3.2 and 3.1 have conflicts, I'll look in a moment

**Comment by evgeni** [comment]
and no, I don't see how to apply the same changes to 3.1, someone else who understands them needs to take a look and do a sperate PR against 3.1 please

**Pull request merged**
